shared-bindings/framebufferio/FramebufferDisplay.c

```diff
@@ -0,0 +1,452 @@
+#include "shared-bindings/microcontroller/Pin.h"
+#include "shared-bindings/util.h"
+#include "shared-module/displayio/__init__.h"
+#include "supervisor/shared/translate.h"
+
+//| .. currentmodule:: framebufferio
+//|
+//| :class:`FramebufferDisplay` -- Manage updating a display with framebuffer in RAM
+//| ================================================================================
+//|
+//| This initializes a display and connects it into CircuitPython. Unlike other
+//| objects in CircuitPython, Display objects live until `displayio.release_displays()`
+//| is called. This is done so that CircuitPython can use the display itself.
+//|
+//| .. class:: FramebufferDisplay(framebuffer, *, width, height, colstart=0, rowstart=0, rotation=0, color_depth=16, grayscale=False, pixels_in_byte_share_row=True, bytes_per_cell=1, reverse_pixels_in_byte=False, backlight_pin=None, brightness=1.0, auto_brightness=False, auto_refresh=True, native_frames_per_second=60)
+//|
+//|   Create a Display object with the given framebuffer (a buffer, array, ulab.array, etc)
+//|
+//|   :param framebuffer: The framebuffer that the display is connected to
+//|   :type framebuffer: any core object implementing the framebuffer protocol
+//|   :param callback: Function or bound method to call when the framebuffer has been updated.  The callback receives one argument, the framebuffer object.
+//|   :param int width: Width in pixels
+//|   :param int height: Height in pixels
+//|   :param int colstart: The index if the first visible column
+//|   :param int rowstart: The index if the first visible row
+//|   :param int rotation: The rotation of the display in degrees clockwise. Must be in 90 degree increments (0, 90, 180, 270)
+//|   :param int color_depth: The number of bits of color per pixel transmitted. (Some displays
+//|       support 18 bit but 16 is easier to transmit. The last bit is extrapolated.)
+//|   :param bool grayscale: True if the display only shows a single color.
+//|   :param bool pixels_in_byte_share_row: True when pixels are less than a byte and a byte includes pixels from the same row of the display. When False, pixels share a column.
+//|   :param int bytes_per_cell: Number of bytes per addressable memory location when color_depth < 8. When greater than one, bytes share a row or column according to pixels_in_byte_share_row.
+//|   :param bool reverse_pixels_in_byte: Reverses the pixel order within each byte when color_depth < 8. Does not apply across multiple bytes even if there is more than one byte per cell (bytes_per_cell.)
+//|   :param bool reverse_pixels_in_byte: Reverses the pixel order within each byte when color_depth < 8. Does not apply across multiple bytes even if there is more than one byte per cell (bytes_per_cell.)
+//|   :param bool reverse_bytes_in_word: Reverses the order of bytes within a word when color_depth == 16
+//|   :param microcontroller.Pin backlight_pin: Pin connected to the display's backlight
```

Comment:
Why do you need these?

Reply:
At least some will apply to other framebuffers we anticipate.  For instance, 1/2/3/4 bytes per pixel are all possible in i.mx lcds from my reading.  I didn't see grayscale / pixels_in_byte_share_row or reverse_xxx being needed yet, can remove them until needed if you desire.